### DISCUSSION TITLE
How do I increase the error threshold for disabling WMS layers

### DISCUSSION THREAD

**Discussion by nf-s** [TerriaJS/terriajs#6933] [Q&A] [answered]
My GeoServer is returning XML with 200 status code as a PNG

**Comment by nf-s** [comment, marked as answer]
Great question @alexgleith

There are four options you can use to tweak this behaviour

- `thresholdBeforeDisablingItem` - The number of tile failures before disabling the item.
- `treat403AsError` - Indicates whether a 403 response code when requesting a tile  should be treated as an error. If false, 403s are assumed to just be  missing tiles and need not be reported to the user.
- `treat404AsError` - Indicates whether a 404 response code when requesting a tile  should be treated as an error. If false, 404s are assumed to just be  missing tiles and need not be reported to the user.
- `ignoreUnknownTileErrors` - A flag indicating whether non-specific (no HTTP status code)  tile errors should be ignored. This is a last resort, for dealing with  odd cases such as data sources that return non-images (eg XML) with a  200 status code. No error messages will be shown to the user.

These properties can be set using the `tileErrorHandlingOptions` object on your WMS config - for example

```json
{
  "type": "wms",
  "name": "Weathering Intensity",
  "url": "https://ows.services.dea.ga.gov.au/",
  "layers": "weathering_intensity",
  "tileErrorHandlingOptions": {
    "ignoreUnknownTileErrors": true
  }
}
```

More info can be found here - https://docs.terria.io/guide/connecting-to-data/catalog-type-details/wms/#tileerrorhandlingtraits

**Comment by alexgleith** [comment]
Just noticing that we see these layers being dropped out only in `2D` and not in `3D`. Is it possible that the two paths, between Leaflet and Cesium are handling the `200` fail errors differently?

We're about to test using the `thresholdBeforeDisablingItem` parameter... will update on how it goes.

**Reply by nf-s** [reply]
Can you please post a screenshot of your network tab for both 2D and 3D mode?

Leaflet (2D) and Cesium (3D) do make requests differently, so you may see different results when using `thresholdBeforeDisablingItem`

**Reply by alexgleith** [reply]
I looked into it a lot. Cesium definitely handles the failures differently... just silently ignores them.

Leaflet renders the broken PNG image thing and calls it a fail.

I've increased thresholdBeforeDisablingItem and that worked a bit, but now we're aggressively ignoring errors, which means we get no layer disabling but sometimes lots of broken PNGs.

The root cause is GeoServer, which is our issue.

Thanks for your help @nf-s !
